### PR TITLE
fix(map_loader): map_loader package not working in UTM coordinates

### PR DESCRIPTION
**Pull request by meliketanrikulu**
## Description

In the map_loader package, lanelet map is only displayed in MGRS coordinates. It should also be able to work in UTM coordinates. This PR adds an option to work with UTM coordinates to the map_loader package.
By setting the lanelet2_map_projector_type parameter to "UTM" or "MGRS" you will be able to choose which coordinate plane to work in.
## Related links

Fixes: #607 

## Notes for reviewers

You should change the lanelet2_map_projector_type parameter to UTM in the map.launch.py file in the tier4_map_launch package. You should specify the latitude longitude values of the starting point of the hd map you use in the lanelet2_map_loader.param.yaml file in the map_loader package. You can see the results in rviz

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp [hunk 1]
```diff
@@ -49,9 +50,22 @@ Lanelet2MapLoaderNode::Lanelet2MapLoaderNode(const rclcpp::NodeOptions & options
...
   lanelet::ErrorMessages errors{};
-  lanelet::projection::MGRSProjector projector{};
-  lanelet::LaneletMapPtr map = lanelet::load(lanelet2_filename, projector, &errors);
+  lanelet::LaneletMapPtr map;
+  if (lanelet2_map_projector_type == "MGRS") {
+    lanelet::projection::MGRSProjector projector{};
+    map = lanelet::load(lanelet2_filename, projector, &errors);
+  } else if (lanelet2_map_projector_type == "UTM") {
+    double map_origin_lat = this->declare_parameter("latitude", 0.0);
+    double map_origin_lon = declare_parameter("longitude", 0.0);
```
Comment (xmfcx) [inline map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp:61]:
```suggestion
    double map_origin_lon = this->declare_parameter("longitude", 0.0);
```

**Comment by xmfcx** [comment]
Thanks for your contribution, it looks good so far!

Does this have a related issue? If so please add it to the description with a remark like: closes: #X

For similar keywords [see here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Please fill in the required fields in the PR template.

**Comment by meliketanrikulu** [comment]
I updated my branch. @xmfcx  Can you check again. Thanks.

**Comment by brkay54** [comment]
Thank you for contributions, I reviewed the code, it looks fine. But I need to check that the changes in map_loader affect the other modules or not? Did you test it? Please share more information in **Notes for reviewers** and **Tests performed**.

**Review by brkay54 — APPROVED** [review]
I tested this PR with planning_simulator, all the modules included planning simulator works fine. Please add the options into `map.launch.py` file.

**Comment by meliketanrikulu** [comment]
@xmfcx can you merge this PR ?

**Review by xmfcx — DISMISSED** [review]

**Review by brkay54 — APPROVED** [review]

**Review by brkay54 — APPROVED** [review]

**Review by xmfcx — APPROVED** [review]

**Pull request merged**